### PR TITLE
Environmental variables resolving

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
**Stack**:
- #1921
- #1911
- #1910 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Comment by MaksymilianDemitraszek** [comment]
Closes #1475 

## Introduced changes



- Env variables are supported in configuration files

## Checklist



- [] Linked relevant issue
- [] Updated relevant documentation
- [] Added relevant tests
- [] Performed self-review of the code
- [] Added changes to `CHANGELOG.md`

**Review by THenry14 — APPROVED** [review]
🔥

File: crates/configuration/src/lib.rs [hunk 1]
```diff
@@ -88,13 +89,49 @@ pub fn load_package_config<T: PackageConfig + Default>(
...
+fn resolve_env_variables(config: serde_json::Value) -> Result<serde_json::Value> {
+    match config {
+        serde_json::Value::Object(map) => {
+            let val = map
+                .into_iter()
+                .map(|(k, v)| -> Result<(String, serde_json::Value)> {
+                    Ok((k, resolve_env_variables(v)?))
+                })
+                .collect::<Result<Vec<(String, serde_json::Value)>>>()?;
+            Ok(serde_json::Value::Object(val.into_iter().collect()))
```
Comment (Draggu) [inline crates/configuration/src/lib.rs:109]:
```suggestion
                .collect::<Result<serde_json::Map<String, serde_json::Value>>>()?;
            Ok(serde_json::Value::Object(val))
```

**Review by Draggu — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,11 +7,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Cast
+
+#### Added
+- Providing configuration data with env variables
```
Comment (piotmag769) [inline CHANGELOG.md:13]: You could link the future docs section

**Review by piotmag769 — APPROVED** [review]

File: crates/configuration/src/lib.rs (hunk 1 above)
Comment (MaksymilianDemitraszek) [reply crates/configuration/src/lib.rs:109]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply CHANGELOG.md]: Done

**Pull request merged**